python/ray/data/_internal/block_batching/iter_batches.py

```diff
@@ -162,6 +161,7 @@ def _async_iter_batches(
             batch_format=batch_format,
             collate_fn=collate_fn,
             num_threadpool_workers=prefetch_batches,
+            metrics_tag=metrics_tag,
```

Comment:
(not blocking this PR) Does it make more sense to say this metrics_tag in DatasetStats? So we don't need to pass them around together